Compilation fails with GCC 14

When building with gcc 14, casacore fails to compile `mirlib/uvio.c` because of a mismatch between int pointer types `(int8 *)`, required as the third argument of the function `rdhdl_c`, and `(off_t *)` .

```log
[   87s] FAILED: mirlib/CMakeFiles/casa_mirlib.dir/uvio.c.o 
[   87s] /usr/bin/gcc -DCFITSIO_VERSION_MAJOR=4 -DCFITSIO_VERSION_MINOR=300 -DHAVE_DYSCO -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_HDF5 -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=8 -DWCSLIB_VERSION_MINOR=2 -Dcasa_mirlib_EXPORTS -I/home/abuild/rpmbuild/BUILD/casacore-3.5.0 -I/home/abuild/rpmbuild/BUILD/casacore-3.5.0/build -I/usr/include/cfitsio -fcx-fortran-rules -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fopenmp -pthread -O2 -g -DNDEBUG -fPIC -MD -MT mirlib/CMakeFiles/casa_mirlib.dir/uvio.c.o -MF mirlib/CMakeFiles/casa_mirlib.dir/uvio.c.o.d -o mirlib/CMakeFiles/casa_mirlib.dir/uvio.c.o -c /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/uvio.c
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/uvio.c: In function ‘uvopen_c’:
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/uvio.c:794:29: error: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
[   87s]   794 |       rdhdl_c(*tno,"vislen",&(uv->offset),hsize_c(uv->item));
[   87s]       |                             ^~~~~~~~~~~~~
[   87s]       |                             |
--
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/miriad.h:182:51: note: expected ‘int8 *’ {aka ‘long long int *’} but argument is of type ‘off_t *’ {aka ‘long int *’}
[   87s]   182 | void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
[   87s]       |                                             ~~~~~~^~~~~
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/uvio.c:814:26: error: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
[   87s]   814 |     rdhdl_c(*tno,"ncorr",&(uv->corr_flags.offset),-1);
[   87s]       |                          ^~~~~~~~~~~~~~~~~~~~~~~~
[   87s]       |                          |
--
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/miriad.h:182:51: note: expected ‘int8 *’ {aka ‘long long int *’} but argument is of type ‘off_t *’ {aka ‘long int *’}
[   87s]   182 | void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
[   87s]       |                                             ~~~~~~^~~~~
[   87s] /home/abuild/rpmbuild/BUILD/casacore-3.5.0/mirlib/uvio.c:815:27: error: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
[   87s]   815 |     rdhdl_c(*tno,"nwcorr",&(uv->wcorr_flags.offset),-1);
[   87s]       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~
[   87s]       |                           |
```

--- a/mirlib/uvio.c
+++ b/mirlib/uvio.c
@@ -370,6 +370,20 @@
 
 #define MYABS(x) ( (x) > 0 ? (x) : -(x) )
 
+/**************************************************************************
+*** GCC 14 detects problematic pointer passing:
+***
+***    In file included from casacore/mirlib/uvio.c:288:
+***    casacore/mirlib/miriad.h:182:51: note: expected ‘int8 *’ {aka ‘long long int *’} but argument is of type ‘off_t *’ {aka ‘long int *’}
+***    182 | void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
+***        |                                             ~~~~~~^~~~~
+***
+***************************************************************************/
+#define RDHDL_C( TNO, KEYWORD, VALUE, DEFVAL ) \
+    { int8 value;                              \
+      rdhdl_c( TNO, KEYWORD, &value, DEFVAL);  \
+      *VALUE = value; }
+
 /*----------------------------------------------------------------------*/
 /*									*/
 /*	Types and static variables.					*/
@@ -746,7 +760,7 @@ void uvopen_c(int *tno,Const char *name,Const char *status)
 #ifdef MIR4
     /* figure out if to read old MIR3 or new MIR4 */
 #if true
-      rdhdl_c(*tno,"vislen",&(uv->max_offset),hsize_c(uv->item));
+      RDHDL_C(*tno,"vislen",&(uv->max_offset),hsize_c(uv->item));
 #else
       int old_vislen;
       rdhdi_c(*tno,"vislen",&old_vislen,hsize_c(uv->item));
@@ -791,7 +805,7 @@ void uvopen_c(int *tno,Const char *name,Const char *status)
 #ifdef MIR4
     /* figure out if to read old MIR3 or new MIR4 */
     if (1) {
-      rdhdl_c(*tno,"vislen",&(uv->offset),hsize_c(uv->item));
+      RDHDL_C(*tno,"vislen",&(uv->offset),hsize_c(uv->item));
     } else {
       int old_vislen;
       rdhdi_c(*tno,"vislen",&old_vislen,hsize_c(uv->item));
@@ -811,8 +825,8 @@ void uvopen_c(int *tno,Const char *name,Const char *status)
     rdhda_c(*tno,"obstype",line,"",MAXLINE);
     if(!strcmp(line,"autocorrelation"))		uv->flags |= UVF_AUTO;
     else if(!strcmp(line,"crosscorrelation"))	uv->flags |= UVF_CROSS;
-    rdhdl_c(*tno,"ncorr",&(uv->corr_flags.offset),-1);
-    rdhdl_c(*tno,"nwcorr",&(uv->wcorr_flags.offset),-1);
+    RDHDL_C(*tno,"ncorr",&(uv->corr_flags.offset),-1);
+    RDHDL_C(*tno,"nwcorr",&(uv->wcorr_flags.offset),-1);
     if(uv->corr_flags.offset < 0 || uv->wcorr_flags.offset < 0)
       BUG('f',"Cannot append to uv file without 'ncorr' and/or 'nwcorr' items");
 

